### PR TITLE
Modal: hide header/footer border line unless content is scrollable

### PR DESCRIPTION
**Pull request by ausnewland11**
## Description

Update to the modal component to hide header and footer border lines unless content is scrollable

References [#2733](https://github.com/trimble-oss/modus-web-components/issues/2733)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: stencil-workspace/src/components/modus-modal/modus-modal.tsx [hunk 1]
```diff
@@ -133,8 +136,16 @@ export class ModusModal {
   }
 
   componentDidRender() {
-    if (this.modalContentRef && this.startTrapRef)
+    if (this.modalContentRef && this.startTrapRef) {
       this.focusWrapping = new ModalFocusWrapping(this.modalContentRef, this.startTrapRef);
+    }
+    this.checkContentScrollable();
```
Comment (cjwinsor) [inline stencil-workspace/src/components/modus-modal/modus-modal.tsx:142]: @ausnewland11 Can you make sure that if the content (slot) is updated and the state changes in a way that would toggle `isContentScrollable` that the modal is updated accordingly?

**Review by cjwinsor — CHANGES_REQUESTED** [review]

File: stencil-workspace/src/components/modus-modal/modus-modal.tsx (hunk 1 above)
Comment (ausnewland11) [reply stencil-workspace/src/components/modus-modal/modus-modal.tsx:142]: @cjwinsor Basing it off the docs [here ](https://modus.trimble.com/components/web/modals/styles/), the modal will be scrollable when the height of the dialog is past it's maximum. So the content/slot will contribute to that and the component re-renders anytime there is a change which will toggle `isContentScrollable` to update. I've written a test to simulate this

Comment (prashanth-offcl) [reply stencil-workspace/src/components/modus-modal/modus-modal.tsx:142]: Hi @ausnewland11, I just tested the scenario Chris mentioned (by dynamically adding content using timeout) and the scrollable class is not added. Can you check it?

Comment (ausnewland11) [reply stencil-workspace/src/components/modus-modal/modus-modal.tsx:142]: @prashanth-offcl Did you do this testing in the browser or by running the component locally?

Comment (prashanth-offcl) [reply stencil-workspace/src/components/modus-modal/modus-modal.tsx:142]: @ausnewland11 I first tried in the deploy preview and ran the component locally as well.

Comment (ausnewland11) [reply stencil-workspace/src/components/modus-modal/modus-modal.tsx:143]: @prashanth-offcl I've pushed an update. Could you please test again now

Comment (prashanth-offcl) [reply stencil-workspace/src/components/modus-modal/modus-modal.tsx:143]: @ausnewland11 This is the test code I'm using and I can still reproduce it. can you try with this once?

```
<modus-button id="btn-modal" color="primary">Open modal</modus-button>
<modus-modal
  header-text="Modal title"
  primary-button-text="Save changes"
  secondary-button-text="Sweet!"
  primary-button-aria-label="Save changes"
  secondary-button-aria-label="Sweet">
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Eligendi voluptatem, deleniti laborum alias temporibus
    fuga iste tempora. Consequatur, aliquid, deleniti commodi ut eos rem, provident possimus iusto pariatur maiores
    autem!
  </p>
</modus-modal>

<script>
  document.querySelector('#btn-modal').addEventListener('buttonClick', () => {
    document.querySelector('modus-modal').open();
  });
  window.setTimeout(() => {
    const modal = document.querySelector('modus-modal');
    let html = '';
    for (let i = 0; i < 40; i++) {
      html += `<p>
                Lorem ipsum dolor sit amet consectetur adipisicing elit. Eligendi voluptatem, deleniti laborum alias temporibus
                fuga iste tempora. Consequatur, aliquid, deleniti commodi ut eos rem, provident possimus iusto pariatur maiores
                autem!
              </p>`;
    }
    modal.innerHTML = html;
  }, 3000);
</script>
```

Comment (ausnewland11) [reply stencil-workspace/src/components/modus-modal/modus-modal.tsx:148]: @prashanth-offcl Thank you for sharing your testing code, with that I was able to see what you were mentioning. I've added a resizeObserver that now works and adds the `Scrollable `class dynamically. Thank you for your patience on this

**Review by prashanth-offcl — APPROVED** [review]

File: stencil-workspace/src/components/modus-modal/modus-modal.tsx (hunk 1 above)
Comment (prashanth-offcl) [reply stencil-workspace/src/components/modus-modal/modus-modal.tsx:154]: @ausnewland11 You're welcome. It is working nice now :)

**Pull request merged**